packages/grafana-data/src/types/data.ts

```diff
@@ -101,6 +101,11 @@ export interface QueryResultMeta {
   limit?: number; // used by log models and loki
   json?: boolean; // used to keep track of old json doc values
   instant?: boolean;
+
+  /**
+   * Indicates a time range that the data source was queried with.
+   */
+  timeRange?: TimeRange;
```

Comment:
```suggestion
  queryTimeRange?: AbsoluteTimeRange;
```

The TimeRange object is not great for serialization 😬 

Also I think we should make it clear that this with the query time range -- the data range is already in the data :)